cirq-core/cirq/work/sampler.py

```diff
@@ -294,10 +294,9 @@ async def run_batch_async(
         See docs for `cirq.Sampler.run_batch`.
         """
         params_list, repetitions = self._normalize_batch_args(programs, params_list, repetitions)
-        return [
-            await self.run_sweep_async(circuit, params=params, repetitions=repetitions)
-            for circuit, params, repetitions in zip(programs, params_list, repetitions)
-        ]
+        return await duet.pstarmap_async(
```

Comment:
Can you add a test that demonstrates the list of programs are all sent without waiting on any to complete? (This should fail under the former implementation). Thanks!